src/status_im/chat/models.cljs

```diff
@@ -101,34 +103,30 @@
               (or (get (:chats db) chat-id)
                   (create-new-chat chat-id cofx))
               chat-props)]
-
     {:db            (update-in db [:chats chat-id] merge chat)
      :data-store/tx [(chats-store/save-chat-tx chat)]}))
 
 (fx/defn add-public-chat
   "Adds new public group chat to db & realm"
   [cofx topic]
-  (upsert-chat cofx
-               {:chat-id                        topic
-                :is-active                      true
-                :name                           topic
-                :group-chat                     true
-                :contacts                       #{}
-                :public?                        true
-                :might-have-join-time-messages? true
-                :unviewed-messages-count        0
-                :loaded-unviewed-messages-ids   #{}}))
-
-(fx/defn add-group-chat
-  "Adds new private group chat to db & realm"
-  [cofx chat-id chat-name admin participants]
-  (upsert-chat cofx
-               {:chat-id     chat-id
-                :name        chat-name
-                :is-active   true
-                :group-chat  true
-                :group-admin admin
-                :contacts    participants}))
+  (if config/use-status-go-protocol?
+    {:json-rpc/call [{:method "status_joinPublicChat"
```

Comment:
why json-rpc ?

Reply:
what do you mean by that? why do we export API as json rpc or why do I use this effect name?

Reply:
why do we export API as json rpc . sorry

Reply:
(1) it is more modular on status-go side
(2) it is more universal if someone wants to create a 3rd party chat UI, they don't need to rely on native code, they can just export this JSON api to `localhost` and make requests.
(3) also easier to debug separately, you only need status-go and `curl` for that.

Reply:
ok i think i know the answer :)